containers-tests/tests/intmap-strictness.hs

```diff
@@ -76,6 +82,70 @@ pInsertLookupWithKeyValueStrict f k v m
                      not (isBottom $ M.insertLookupWithKey (const3 1) k bottom m)
     | otherwise    = isBottom $ M.insertLookupWithKey (apply3 f) k bottom m
 
+pFromAscListLazy :: [Int] -> Bool
+pFromAscListLazy ks = not . isBottom $ M.fromAscList elems
+  where
+    ks' = map head . group . sort $ ks
+    elems = [(k, v) | k <- ks', v <- [undefined, ()]]
+
+pFromAscListStrict :: [Int] -> Bool
+pFromAscListStrict ks
+    | null ks   = not . isBottom $ M.fromAscList elems
+    | otherwise = isBottom $ M.fromList elems
+  where
+    ks' = map head . group . sort $ ks
+    elems = [(k, v) | k <- ks', v <- [undefined, undefined, ()]]
+
+------------------------------------------------------------------------
+-- check for extra thunks
+
+tExtraThunksM :: Test
+tExtraThunksM = testGroup "IntMap.Strict - extra thunks" $
```

Comment:
What are extra thunks? Maybe add a brief explanation and/or a reference to a definition or discussion.